x-pack/plugins/actions/server/builtin_action_types/lib/get_oauth_jwt_access_token.ts

```diff
@@ -72,6 +72,9 @@ export const getOAuthJwtAccessToken = async ({
       keyId: jwtKeyId,
     });
 
+    // Save the time before requesting token so we can use it to calculate expiration
+    const requestTokenStart = Date.now();
```

Comment:
using the time before requesting the token to calculate the expiration time of the token. previously, the expiration was calculated using a `Date.now()` inside `connectorTokenClient.updateOrReplace`, which occurs after the token has been requested and received. If there's any delay in processing between when the token was requested and when it was persisted to the `connector_token` saved object, it was possible to set an expiration that was past when the token would actually expire.